src/router.js

```diff
@@ -19,11 +20,15 @@ export const ROUTE_NAME_SIGNUP = 'Signup';
 export const ROUTE_NAME_STATS = 'Stats';
 export const ROUTE_NAME_STATS_SEASON = 'StatsBySeason';
 
-const mustBeAuthenticated = (to, from, next) => {
+const mustBeAuthenticated = async (to, from, next) => {
   if (store.state.auth.authenticated) {
     return next();
   }
-  return next('/login');
+  await store.dispatch('getReturningUser');
```

Comment:
This action is synchronous
```suggestion
store.dispatch('getReturningUser');
```